torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -285,6 +285,154 @@ void ThreadPredicateMap::updateBitSet(const Expr* expr) {
   }
 }
 
+namespace {
+
+//! A simple backward data flow pass:
+//!  This pass propagates information backward to annotate "redundant use
+//!  chain"'s.
+//! The reason this is needed is that if we have a chain of register-to-register
```

Comment:
Is this propagation analysis strictly for register-to-register expr chains? If so, why there's a propagation rule for shared memory tensors?

Reply:
Shared memory part is trying to include analysis for BID as well so this info will be there when we need it, though I don't think we have been exercising global sync insertion too much at the moment.

Reply:
OK, but the comment seems confusing then?

Reply:
Ah, yes it seems outdated. Just updated it. Thanks!